server/services/prismic.js

```diff
@@ -241,43 +235,37 @@ function convertStringToNumber(string: string): number {
   return Number(string.replace(/\D/g, ''));
 }
 
-async function getResults(page: number, type: string): Promise<any> { // TODO make type its own enumerable thing}
-  switch (type) {
-    case 'exhibition':
-      const exhibitions = await getExhibitions(page);
+function convertPrismicResultsToPaginatedResults(prismicResults: Object): (results: PaginatedResultsType) => PaginatedResults {
+  const currentPage = prismicResults && prismicResults.page;
+  const pageSize = prismicResults && prismicResults.results_per_page;
+  const totalResults = prismicResults && prismicResults.total_results_size;
+  const totalPages = prismicResults && prismicResults.total_pages;
+  const pagination = PaginationFactory.fromList(List(prismicResults.results), parseInt(totalResults, 10) || 1, parseInt(currentPage, 10) || 1, pageSize || 1);
 
-      return exhibitions;
-    case 'event':
-      const events = await getEvents(page);
-      return events;
-  }
-}
-
-export async function getPaginatedResults(page: number, type: string): Promise<PaginatedResults> { // TODO make type its
-  const allResults = await getResults(page, type);
-  const currentPage = allResults && allResults.page;
-  const pageSize = allResults && allResults.results_per_page;
-  const totalResults = allResults && allResults.total_results_size;
-  const totalPages = allResults && allResults.total_pages;
-  const pagination = PaginationFactory.fromList(List(allResults.results), parseInt(totalResults, 10) || 1, parseInt(page, 10) || 1, pageSize || 1);
-
-  const promos = convertResultsToPromos(allResults, type);
-
-  return {
-    currentPage,
-    totalPages,
-    results: promos,
-    pagination
+  return (results) => {
+    return {
+      results,
+      currentPage,
+      pageSize,
+      totalResults,
+      totalPages,
+      pagination
+    };
   };
 }
 
-async function getEvents(page:number = 1, pageSize:number = 40) {
-  const prismic = await getPrismicApi();
-  const events = await prismic.query([
-    Prismic.Predicates.any('document.type', ['events'])
-  ], {page, pageSize}); // TODO: add orderings by first time in times?
+export async function getEventPromos(page: number): Promise<Array<EventPromo>> {
+  const results = await getAllOfType('events', page, '[my.events.times.startDateTime desc]');
```

Comment:
This is the important piece, happy to remove the refactor...
I was just finding it hard to see where the type string was being passed around (`event` vs `events` etc, this way the DocumentType is locked up here and clients don't need to know about it).